graylog2-server/src/main/java/org/graylog2/events/ClusterEventPeriodical.java

```diff
@@ -61,70 +55,126 @@
-                               final MongoConnection mongoConnection,
-                               final NodeId nodeId,
-                               final ObjectMapper objectMapper,
-                               final EventBus serverEventBus,
-                               @Named("cluster_event_bus") final EventBus clusterEventBus) {
+    public ClusterEventPeriodical(final MongoJackObjectMapperProvider mapperProvider,
+                                  final MongoConnection mongoConnection,
+                                  final NodeId nodeId,
+                                  final ObjectMapper objectMapper,
+                                  final EventBus serverEventBus,
+                                  @Named("cluster_event_bus") final EventBus clusterEventBus) {
         this(JacksonDBCollection.wrap(prepareCollection(mongoConnection), ClusterEvent.class, String.class, mapperProvider.get()),
                 nodeId, objectMapper, serverEventBus, clusterEventBus);
     }
 
-    ClusterEventService(final JacksonDBCollection<ClusterEvent, String> dbCollection,
-                        final NodeId nodeId,
-                        final ObjectMapper objectMapper,
-                        final EventBus serverEventBus,
-                        final EventBus clusterEventBus) {
+    ClusterEventPeriodical(final JacksonDBCollection<ClusterEvent, String> dbCollection,
+                           final NodeId nodeId,
+                           final ObjectMapper objectMapper,
+                           final EventBus serverEventBus,
+                           final EventBus clusterEventBus) {
         this.nodeId = checkNotNull(nodeId);
         this.dbCollection = checkNotNull(dbCollection);
         this.objectMapper = checkNotNull(objectMapper);
         this.serverEventBus = checkNotNull(serverEventBus);
-        this.clusterEventBus = checkNotNull(clusterEventBus);
 
-        this.clusterEventBus.register(this);
+        checkNotNull(clusterEventBus).register(this);
     }
 
     @VisibleForTesting
     static DBCollection prepareCollection(final MongoConnection mongoConnection) {
         final DB db = mongoConnection.getDatabase();
 
-        DBCollection coll;
-        if (!db.collectionExists(COLLECTION_NAME)) {
-            final DBObject options = BasicDBObjectBuilder.start()
-                    .add("capped", true)
-                    .add("size", 32768)
-                    .get();
-
-            try {
-                coll = db.createCollection(COLLECTION_NAME, options);
-            } catch (MongoException e) {
-                // Error code 48 == "collection already exists"
-                if(e.getCode() == 48) {
-                    coll = db.getCollection(COLLECTION_NAME);
+        DBCollection coll = db.getCollection(COLLECTION_NAME);
+
+        coll.createIndex(DBSort.desc("timestamp"));
+        coll.createIndex(DBSort.asc("producer"));
+        coll.createIndex(DBSort.asc("consumers"));
```

Comment:
Does this work pre-mongodb 2.6?

The query below is a and-query with a sort on timestamp. I think this should be a single index (timestamp desc, producers asc, consumers asc) instead.